plugins/BEdita/Core/src/Model/Action/RemoveAssociatedAction.php

```diff
@@ -58,10 +59,20 @@ protected function diff(EntityInterface $entity, array $relatedEntities)
      * @param \Cake\Datasource\EntityInterface $entity Source entity.
      * @param \Cake\Datasource\EntityInterface|\Cake\Datasource\EntityInterface[]|null $relatedEntities Related entity(-ies).
      * @return int|false Number of updated relationships, or `false` on failure.
+     * @throws \ForbiddenException Throws an exception if trying to remove relation between administrator user and administrator role.
      * @throws \RuntimeException Throws an exception if an unsupported association is passed.
      */
     protected function update(EntityInterface $entity, $relatedEntities)
     {
+        if (is_array($relatedEntities)) {
+            $ids = array_keys($relatedEntities);
+            if ((in_array(\BEdita\Core\Model\Table\RolesTable::ADMIN_ROLE, $ids) && $entity instanceof \BEdita\Core\Model\Entity\Role)
+                ||
+                (in_array(\BEdita\Core\Model\Table\UsersTable::ADMIN_USER, $ids) && $entity instanceof \BEdita\Core\Model\Entity\User)) {
+                throw new ForbiddenException(__d('bedita', 'Could not update relationship for users/roles for ADMIN_USER and ADMIN_ROLE'));
```

Comment:
Again, I'd prefer this logic to be moved in `RolesUsersTable::beforeDelete()` and raise an exception from there.

Here it is not only because of "coding style", but it's also because not all cases are covered by this check: an invocation to `SetAssociatedAction` would replace all existing links, so by issuing a request on `PATCH /users/1/relationships/roles` with an empty array of data one would still be able to remove user 1 from administrators.